code/modules/awaymissions/corpse.dm

```diff
@@ -177,20 +177,21 @@
 		if(id_icon)
 			W.icon_state = id_icon
 		if(id_access)
-			var/datum/job/jobdatum
+			var/list/jobdatum_access = list()
+			var/datum/job/J
 			for(var/jobtype in typesof(/datum/job))
-				var/datum/job/J = new jobtype
+				J = new jobtype
 				if(J.title == id_access)
-					jobdatum = J
+					jobdatum_access = J.get_access()
+					qdel(J)
 					break
-			if(jobdatum)
-				W.access = jobdatum.get_access()
-			else
+				qdel(J)
```

Comment:
I don't like this of calling qdel both times(never for the same datum but still). But couldn't figure out a way to do it just once without leaving useless job datums around.